src/jsx-parser.ts

```diff
@@ -256,9 +266,12 @@ export class JSXParser extends Parser {
         }
 
         this.scanner.throwUnexpectedToken();
+
+        /* istanbul ignore next */
+        throw new Error('Unreachable');
```

Comment:
If you add a type annotation to `throwError` of type `never`, and then have `throwUnexpectedToken` actually return the call to `throwError`, you should get it working.